staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/validation.go

```diff
@@ -251,6 +252,26 @@ func (s *Validator) validateExpressions(ctx context.Context, fldPath *field.Path
 			continue
 		}
 		if evalResult != types.True {
+			tryMessage := true
+			if len(rule.MessageExpression) != 0 {
+				evalResult, evalDetails, err := compiled.MessageExpression.ContextEval(ctx, activation)
+				if evalDetails == nil {
+					klog.Warningf("runtime cost could not be calculated for message expression %q")
+				}
+				if err != nil {
+					klog.Warningf("error evaluating message expression, falling back: %v", err.Error())
+				} else {
+					messageStr, ok := evalResult.Value().(string)
+					if !ok {
+						klog.Warningf("could not could message expression result to string, falling back: %v", evalResult.Value())
```

Comment:
The message: `could not resolve messageExpression to string`?

Reply:
And stop here if `!ok`? We don't want to append err if cannot resolve to string right?

Reply:
Right - I've re-worked the error paths (and fixed the typo in the message, terribly sorry about that).